app/app.go

```diff
@@ -204,11 +231,19 @@ func (a *App) PushHandler(push *msg.Push) {
 }
 
 // getLocalChain returns an instance of the blockchain.
-func getLocalChain() *blockchain.BlockChain {
+func getLocalChain(user *User) *blockchain.BlockChain {
```

Comment:
It seems like the function is overloaded with two separate situations:
- we want to start a new chain with a genesis block
- we want to continue an old chain by restoring from storage

We should have two separate functions for these two situations.